compiler/rustc_trait_selection/src/traits/select/mod.rs

```diff
@@ -1038,19 +1038,15 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
         it.for_each(|o| o.recursion_depth = cmp::max(min_depth, o.recursion_depth) + 1);
     }
 
-    /// Checks that the recursion limit has not been exceeded.
-    ///
-    /// The weird return type of this function allows it to be used with the `try` (`?`)
-    /// operator within certain functions.
-    fn check_recursion_limit<T: Display + TypeFoldable<'tcx>, V: Display + TypeFoldable<'tcx>>(
+    fn check_recursion_depth<T: Display + TypeFoldable<'tcx>>(
```

Comment:
infcx is very sensitive to changes in hot code paths, maybe this change caused it? Can you do a PR without this change (duplicating for your use case) and see how it goes? Or maybe just inline(always) check_recursion_limit?

Reply:
Let me open a pr adding inline(always) and see how that goes. I don't think anything other than this code caused the regression

Reply:
Opened #88962.